### ISSUE TITLE
CrashLoopBackOff occurred when deploy test cluster

### ISSUE THREAD

**Issue by sukkiCat** [pingcap/tidb-operator#4274]
## Bug Report

**What version of Kubernetes are you using?**

```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:41:28Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:32:41Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
```
**What version of TiDB Operator are you using?**

```
Defaulted container "tidb-scheduler" out of: tidb-scheduler, kube-scheduler
TiDB Operator Version: version.Info{GitVersion:"v1.2.4", GitCommit:"684ce274555ac739a58c6ed07ff673a2d98a5429", GitTreeState:"clean", BuildDate:"2021-10-21T03:22:23Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```
**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

```
NAME                 PROVISIONER                RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
standard (default)   k8s.io/minikube-hostpath   Delete          Immediate           false                  8m52s

sukki_qin@cloudshell:~$ kubectl get pvc -n tidb-admin
No resources found in tidb-admin namespace.
```
**What's the status of the TiDB cluster pods?**

```
sukki_qin@cloudshell:~$ kubectl get po -n tidb-admin -o wide
NAME                                       READY   STATUS             RESTARTS        AGE   IP           NODE       NOMINATED NODE   READINESS GATES
tidb-controller-manager-7c79b4567c-lmd7w   1/1     Running            0               10m   172.17.0.3   minikube   <none>           <none>
tidb-scheduler-759b84cd6d-5zpw2            1/2     CrashLoopBackOff   6 (4m18s ago)   10m   172.17.0.4   minikube   <none>           <none>
```
**What did you do?**

I've tried to deploy a test cluster using minikube following the instruction of https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/get-started#%E4%BD%BF%E7%94%A8-minikube-%E5%88%9B%E5%BB%BA-kubernetes-%E9%9B%86%E7%BE%A4
**What did you expect to see?**
all pods are running
**What did you see instead?**
tidb-scheduler pods are failed to start, an error status is `CrashLoopBackOff`.
I've checked the log of this pod, no error found
```
sukki_qin@cloudshell:~$ kubectl logs --namespace tidb-admin tidb-scheduler-759b84cd6d-5zpw2 -c tidb-scheduler
I1123 07:24:14.732038       1 version.go:38] Welcome to TiDB Operator.
I1123 07:24:14.732100       1 version.go:39] TiDB Operator Version: version.Info{GitVersion:"v1.2.4", GitCommit:"684ce274555ac739a58c6ed07ff673a2d98a5429", GitTreeState:"clean", BuildDate:"2021-10-21T03:22:33Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
I1123 07:24:14.732164       1 main.go:61] FLAG: --V="false"
I1123 07:24:14.732175       1 main.go:61] FLAG: --add_dir_header="false"
I1123 07:24:14.732181       1 main.go:61] FLAG: --alsologtostderr="false"
I1123 07:24:14.732187       1 main.go:61] FLAG: --features="AdvancedStatefulSet=false,AutoScaling=false,StableScheduling=true"
I1123 07:24:14.732211       1 main.go:61] FLAG: --log_backtrace_at=":0"
I1123 07:24:14.732220       1 main.go:61] FLAG: --log_dir=""
I1123 07:24:14.732229       1 main.go:61] FLAG: --log_file=""
I1123 07:24:14.732235       1 main.go:61] FLAG: --log_file_max_size="1800"
I1123 07:24:14.732243       1 main.go:61] FLAG: --logtostderr="true"
I1123 07:24:14.732249       1 main.go:61] FLAG: --port="10262"
I1123 07:24:14.732258       1 main.go:61] FLAG: --skip_headers="false"
I1123 07:24:14.732264       1 main.go:61] FLAG: --skip_log_headers="false"
I1123 07:24:14.732270       1 main.go:61] FLAG: --stderrthreshold="2"
I1123 07:24:14.732278       1 main.go:61] FLAG: --v="2"
I1123 07:24:14.732285       1 main.go:61] FLAG: --version="false"
I1123 07:24:14.732290       1 main.go:61] FLAG: --vmodule=""
I1123 07:24:14.807097       1 mux.go:67] start scheduler extender server, listening on 0.0.0.0:10262

```

**Comment by KanShiori** [comment]
duplicated with #4155.

It will be fixed in v1.3.0

**Comment by DanielZhangQD** [comment]
@sukkiCat This has been fixed by https://github.com/pingcap/tidb-operator/pull/4202, you can try with the `nightly` image.
